weather.py

```diff
@@ -46,7 +49,10 @@ def calculate_mean(weather_data):
     Returns:
         A float representing the mean value.
     """
-    pass
+    total = 0
+    for x in weather_data:
+        total += float(x)
```

Comment:
It's generally good practice to use informative variable names and avoid the use of [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)) e.g. `x`, to improve code readability. This probably won't be an issue here, since the code is simple and it's obvious what `x` is, but will become more important when we have lots of variables like `x`, `a`, etc